### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by camfitz**
Cleaned up, ready for pull to master.

This fixes all display issues I've noted so far--
- window scrolls
- ls doesn't overflow the display
- page up, page down, home and end keys can scroll the window

### CODE REVIEW

File: gitnu/app/gitnuterminal.dart [hunk 1]
```diff
@@ -149,6 +181,78 @@ class GitnuTerminal extends Terminal {
...
+        });
+
+        html.write('</div>');
+        writeOutput(html.toString());
+      }
+    };
+
+    // Read contents of current working directory. According to spec, need to
+    // keep calling readEntries() until length of result array is 0. We're
+    // guarenteed the same entry won't be returned again.
```
Comment (soulplant) [inline gitnu/app/gitnuterminal.dart]: typo: guaranteed

File: gitnu/app/lib/terminal-dart/terminal.dart [hunk 2]
```diff
@@ -119,7 +117,7 @@ class Terminal {
       // Function look up
       if (cmds[cmd] is Function) {
         cmds[cmd](cmd, args);
-      } else {
+      } else  {
```
Comment (soulplant) [inline gitnu/app/lib/terminal-dart/terminal.dart]: nit: there's an extra space here

File: gitnu/app/lib/terminal-dart/terminal.dart [hunk 3]
```diff
@@ -37,13 +37,11 @@ class Terminal {
 
     // Trick: Always force text cursor to end of input line.
     cmdLine.onClick.listen((event) => input.value = input.value);
-
+
```
Comment (soulplant) [inline gitnu/app/lib/terminal-dart/terminal.dart]: nit: trailing whitespace

File: gitnu/app/gitnuterminal.dart [hunk 4]
```diff
@@ -149,6 +181,78 @@ class GitnuTerminal extends Terminal {
...
+  }
+  
+  /**
+   * Added sorting of entries.
+   */
+  void lsCommand(String cmd, List<String> args) {
+    
+    void displayFiles(List<Entry> entry) {
+      if (entry.length != 0) {
+
```
Comment (soulplant) [inline gitnu/app/gitnuterminal.dart]: nit: no newline

File: gitnu/app/gitnuterminal.dart [hunk 5]
```diff
@@ -149,6 +181,78 @@ class GitnuTerminal extends Terminal {
...
+    
+    sb.write('<div class="ls-files" style="-webkit-column-count: $colCount;">');
+    return sb;
+  }
+  
+  /**
+   * Added sorting of entries.
+   */
+  void lsCommand(String cmd, List<String> args) {
+
```
Comment (soulplant) [inline gitnu/app/gitnuterminal.dart]: nit: no newline

**Comment by soulplant** [comment]
Don't be afraid to mangle beyond recognition the original Terminal code to appropriate it for your purposes. I'm thinking it could expose an interface for registering commands with it, eg:

```
// Don't inherit from Terminal.
class GitnuTerminal {
  GitnuTerminal(Terminal terminal) {
    terminal.registerCommand('ls', lsCommand);
  }
}
```

**Comment by camfitz** [comment]
Nits resolved. Next commit will feature design change away from treating the existing terminal code as a library.

**Pull request merged**